pkg/api/handlers/mcpgateway/auditlog_test.go

```
@@ -162,6 +172,76 @@ func TestAttributeMCPAuditLogAPIKeyPreservesPrincipalSnapshot(t *testing.T) {
+	}
+	seedMCPAuditLogs(t, gatewayClient, gatewaytypes.MCPAuditLog{
+		CreatedAt: time.Now().UTC(), SourceType: types.AuditLogSourceTypeMCP,
+		UserID: "7", APIKeyID: &created.ID, APIKeyName: "CLI token",
+		MCPFields: &gatewaytypes.MCPAuditLogFields{MCPID: "mcp-1", CallType: "tools/call"},
+	})
+
+	recorder := httptest.NewRecorder()
+	ctx := newAuditLogListContextWithRecorder(t, gatewayClient, "event_type=mcp_call", &user.DefaultInfo{
+		UID: "1", Groups: []string{types.GroupAuthenticated, types.GroupAdmin},
+	}, recorder)
+	ctx.Request.SetPathValue("filter", "api_key_id")
```

golangci-lint (2.12.2)

[error] QF1008: could remove embedded field "Request" from selector

(staticcheck)